cpp/simulations/abm_braunschweig.cpp

```diff
@@ -0,0 +1,1106 @@
+    {
+        Type location_information{};
+        for (auto&& location : sim.get_world().get_locations()) {
+            location_information.push_back(std::make_tuple(location.get_index(), location.get_geographical_location()));
+        }
+        return location_information;
+    }
+};
+
+struct LogPersonInformation : mio::LogOnce {
+    using Type = std::vector<std::tuple<uint32_t, uint32_t, mio::abm::AgeGroup>>;
+    static Type log(const mio::abm::Simulation& sim)
+    {
+        Type person_information{};
+        for (auto&& person : sim.get_world().get_persons()) {
+            person_information.push_back(std::make_tuple(
+                person.get_person_id(), sim.get_world().find_location(mio::abm::LocationType::Home, person).get_index(),
+                person.get_age()));
+        }
+        return person_information;
+    }
+};
+
+mio::IOResult<void> run(const fs::path& result_dir, size_t num_runs, bool save_single_runs = true)
+{
+
+    auto t0               = mio::abm::TimePoint(0); // Start time per simulation
+    auto tmax             = mio::abm::TimePoint(0) + mio::abm::days(10); // End time per simulation
+    auto ensemble_results = std::vector<std::vector<mio::TimeSeries<ScalarType>>>{}; // Vector of collected results
+    ensemble_results.reserve(size_t(num_runs));
+    auto run_idx            = size_t(1); // The run index
+    auto save_result_result = mio::IOResult<void>(mio::success()); // Variable informing over successful IO operations
+
+    // Loop over a number of runs
+    while (run_idx <= num_runs) {
+
+        // Create the sampled simulation with start time t0.
+        auto sim = create_sampled_simulation(t0);
+        //output object
+        mio::History<mio::DataWriterToMemory, LogLocationInformation, LogPersonInformation> history;
+        // Collect the id of location in world.
+        std::vector<int> loc_ids;
+        for (auto& location : sim.get_world().get_locations()) {
+            loc_ids.push_back(location.get_index());
+        }
+        // Advance the world to tmax
+        sim.advance(tmax, history);
+        // TODO: update result of the simulation to be a vector of location result.
+        auto temp_sim_result = std::vector<mio::TimeSeries<ScalarType>>{sim.get_result()};
+        // Push result of the simulation back to the result vector
+        ensemble_results.push_back(temp_sim_result);
+        // Option to save the current run result to file
+        if (save_result_result && save_single_runs) {
+            auto result_dir_run = result_dir / ("abm_result_run_" + std::to_string(run_idx) + ".h5");
+            BOOST_OUTCOME_TRY(save_result(ensemble_results.back(), loc_ids, 1, result_dir_run.string()));
+        }
+        ++run_idx;
+    }
+    BOOST_OUTCOME_TRY(save_result_result);
+    return mio::success();
+}
+
+int main(int argc, char** argv)
+{
+    mio::set_log_level(mio::LogLevel::warn);
+
+    std::string result_dir = ".";
```

Comment:
The `input_dir` should also be taken into consideration and passed to the run function.